test/functional/wallet_abandonconflict.py

```diff
@@ -120,6 +120,12 @@ def run_test(self):
         assert_equal(newbalance, balance + Decimal("30"))
         balance = newbalance
 
+        # Check abandoned transaction in listsinceblock
+        listsinceblock = self.nodes[0].listsinceblock()
+        txAB1_listsinceblock = [d for d in listsinceblock['transactions'] if d['txid'] == txAB1]
+        assert_equal(txAB1_listsinceblock[0]['abandoned'], True)
+        assert_equal(txAB1_listsinceblock[1]['abandoned'], True)
```

Comment:
Some ideas.
```diff
-        # Check abandoned transaction in listsinceblock
+        self.log.info("Check abandoned transactions in listsinceblock")
         listsinceblock = self.nodes[0].listsinceblock()
-        txAB1_listsinceblock = [d for d in listsinceblock['transactions'] if d['txid'] == txAB1]
-        assert_equal(txAB1_listsinceblock[0]['abandoned'], True)
-        assert_equal(txAB1_listsinceblock[1]['abandoned'], True)
+        txAB1_listsinceblock = [d for d in listsinceblock['transactions'] if d['txid'] == txAB1 and d['category'] == 'send']
+        for tx in txAB1_listsinceblock:
+            assert_equal(tx['abandoned'], True)
```

Reply:
Great, going to update it! I didn't use the `self.log.info` before because I noticed this test don't use it a lot (maybe a improv to do).